x-pack/plugins/observability_solution/logs_explorer/public/controller/create_controller.ts

```diff
@@ -33,7 +37,7 @@ interface Dependencies {
   plugins: LogsExplorerStartDeps;
 }
 
-type InitialState = LogsExplorerPublicStateUpdate;
+type InitialState = LogsExplorerPublicStateUpdate & { allSelection?: AllDatasetSelection };
```

Comment:
Intersection here so that `LogsExplorerPublicStateUpdate` maintains it's tidy 1:1 with public state.